src/sysusers/sysusers.c

```diff
@@ -126,6 +138,12 @@ typedef struct Context {
 static void context_done(Context *c) {
         assert(c);
 
+#if HAVE_AUDIT
+        if (c->audit_fd >= 0)
+                audit_close(c->audit_fd);
+        c->audit_fd = -EBADF;
+#endif
```

Comment:
please add a static inline helper for this in audit-util.h that either is a NOP or closes the fd,